src/env-inl.h

```diff
@@ -296,6 +296,8 @@ inline void Environment::AssignToContext(v8::Local<v8::Context> context,
 }
 
 inline Environment* Environment::GetCurrent(v8::Isolate* isolate) {
+  if (UNLIKELY(!isolate->InContext())) return nullptr;
+  v8::HandleScope handle_scope(isolate);
```

Comment:
sorry for being blunt, but does this scope (in the callee) persist in the code block containing caller's callsite?

Reply:
This scope is not visible to the caller, if that’s what you’re asking? The main difference is that now, the caller does not require a `HandleScope` on its own anymore.